#! /usr/bin/env python
# Forearm camera configuration
 
PACKAGE='ueye'
import roslib; roslib.load_manifest(PACKAGE)
  
#from driver_base.msg import SensorLevels
from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

#       Name                    Type      Reconfiguration level             Description         					Default  Min   	Max
gen.add("force_streaming",	bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Force the camera to stream images.", 			False)
gen.add("hardware_gamma",	bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Sets the camera hardware gamma correction.", 		True)
gen.add("auto_exposure",	bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Sets the camera exposure duration to automatic.", 		False)
gen.add("exposure_time",	double_t, SensorLevels.RECONFIGURE_RUNNING, "Sets the camera exposure time (ms)", 			85, 	0.06, 	200)
gen.add("zoom", 		int_t,    SensorLevels.RECONFIGURE_RUNNING, "The zoom level. (Global binning)", 			1, 	1, 	4)
gen.add("frame_rate",		double_t, SensorLevels.RECONFIGURE_RUNNING, "Sets the expected triggering rate ", 			30, 	1, 	240)
gen.add("pixel_clock",		int_t,    SensorLevels.RECONFIGURE_RUNNING, "Pixel clock speed in MHz", 				86, 	5, 	86)
gen.add("frame_id",		str_t,    SensorLevels.RECONFIGURE_RUNNING, "The optical camera tf frame set in message headers.", 	"/camera")

exit(gen.generate(PACKAGE, "ueye", "camera"))

